tests/mock_callable_testslide.py

```diff
@@ -164,6 +173,13 @@ def patching_private_functions_with_allow_private(self):
         ).and_assert_called_once()
         t._privatefun()
 
+    @context.example
+    def calling_patched_functions_with_bad_types_raises_TypeError(self):
+        t = TargetStr()
+        self.mock_callable(t, "typedfun").to_return_value("This is patched")
+        with self.assertRaises(TypeError):
```

Comment:
Better use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex here, and assert that the error str matches what we expect. Without it, if there's a TypeError for other reasons (eg: giving more arguments than expected), test will pass, but it won't be actually testing type checking.